src/tree-view/tree-item/utils.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { TreeViewProps } from '../interfaces';
+
+interface TransformTreeItemPropsParams
+  extends Pick<TreeViewProps, 'expandedItems' | 'renderItem' | 'getItemId' | 'getItemChildren'> {
+  item: any;
+  index: number;
+}
+
+export function transformTreeItemProps({
```

Comment:
nit: this function is only used once, inside the tree item - wouldn't it be easier to have the code inline?

before:
```
const { id, isExpandable, isExpanded, children, icon, content, secondaryContent, actions } = transformTreeItemProps({
    item,
    index,
    expandedItems,
    renderItem,
    getItemId,
    getItemChildren,
  });
```

after:
```
const { icon, content, secondaryContent, actions } = renderItem(item, index);
const id = getItemId(item, index);
const children = getItemChildren(item, index) || [];
const isExpandable = children.length > 0;
const isExpanded = isExpandable && expandedItems.includes(itemId);
```